File: API/API/Features/v1/WarehouseArea/WarehouseResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CLERP.API.Features.v1.WarehouseArea
{
    /// <summary>
    /// Default response dto for a warehouse
    /// </summary>
    public class WarehouseResponse
    {
        [JsonProperty("id")]
        public Guid Guid { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("shelves")]
        public IEnumerable<WarehouseShelfResponse> Shelves { get; set; }

        [JsonProperty("address")]
        public WarehouseAddressResponse Address { get; set; }
    }
}

File: Client/src/app/_generated/models/warehouse-response.ts
/* tslint:disable */
import { WarehouseShelfResponse } from './warehouse-shelf-response';
import { WarehouseAddressResponse } from './warehouse-address-response';

/**
 * Default response dto for a warehouse
 */
export interface WarehouseResponse {
  id?: string;
  name?: string;
  shelves?: Array<WarehouseShelfResponse>;
  address?: WarehouseAddressResponse;
}
